packages/app/src/app/overmind/namespaces/workspace/actions.ts

```diff
@@ -444,3 +446,53 @@ export const openDefaultItem: Action = ({ state }) => {
 
   state.workspace.openedWorkspaceItem = defaultItem.id;
 };
+
+export const getDependencies: AsyncAction<string | void> = async (
+  { state, effects },
+  value
+) => {
+  state.workspace.loadingDependencySearch = true;
+  const searchResults = await effects.algoliaSearch.searchDependencies(value);
+
+  state.workspace.loadingDependencySearch = false;
+
+  if (!value) {
+    state.workspace.starterDependencies = [...searchResults];
```

Comment:
Just curious why you spread here? Or just to be safe? 😄

Reply:
Honestly because just setting it sometimes workls and sometimes it doesn't ahah

Reply:
Ah, haha, that is confusing 😂 

Are you able to produce like a tiny example of when it works and not? It might be an Overmind bug

Reply:
oh

just messaged you one ahah

Reply:
Oh, I see what you were trying to do now. So `searchResults` is an array that you initially tried to insert on two different state paths, `dependencies` and `starterDependencies`. That does not work, cause now a single value has two different "addresses/paths" in the state tree. So to fix that you made a copy of it, but you actually only need to make one copy. It is enough to copy `starterDependencies` 😄 

But I looked a bit closer and I do not think you need `starterDependencies`? You seem to use it when the search query is back to nothing, but no search query sets the `dependencies` back to normal automatically. So you can:

- Remove `starterDependencies` state
- Remove `setDependencies` action
- And in component only do: `actions.workspace.getDependencies(searchValue);` 

Cause algolia takes care of caching and stuff here I would assume. Anyways, I tried it locally and works as expected. So guess it is a nice tidy up? 😄

Reply:
dude works 10/10

You are hero